src/Iris/Algebra/CMRA.lean

```diff
@@ -21,12 +21,16 @@ class CMRA (α : Type _) extends OFE α where
 
   valid_validN : valid x ↔ ∀ n, validN n x
   validN_succ : validN n.succ x → validN n x
+  validN_op_l : validN n (op x y) → validN n x
+
   assoc : op x (op y z) ≡ op (op x y) z
   comm : op x y ≡ op y x
+
   pcore_l : pcore x = some cx → op cx x ≡ x
   pcore_idem : pcore x = some cx → pcore cx ≡ some cx
-  pcore_mono' : pcore x = some cx → ∃ cy, pcore (op x y) = some (op cx cy)
-  validN_op_l : validN n (op x y) → validN n x
+  pcore_mono : (∃ z, y ≡ op x z) → pcore x = some cx →
+    ∃ cy, pcore y = some cy ∧ ∃ z, cy ≡ op cx z
```

Comment:
I've just re-convinced myself that `pcore_mono'` suffices to prove this statement. Do you want to discuss it on zulip? The version with <= should certainly exist as a lemma.

Reply:
Sure. I'm not that good at juggling multiple equality relations at the same time.

Reply:
Returned old definitions, added the lemma. Had to rename the property since there's a lemma with the same name.